### DISCUSSION TITLE
Patches for librealsense2?

### DISCUSSION THREAD

**Discussion by ethanc8** [Joshua-Riek/ubuntu-rockchip#1094] [General]
We have Intel Realsense cameras, which need some kernel patches in order to work properly. I don't know how easy it will be to apply the patches to this kernel.

**Comment by ethanc8** [comment]
The DKMS package for Ubuntu kernels 5.15.0, 5.19.0, 6.2.0, and 6.5.0 is available at https://librealsense.intel.com/Debian/apt-repo/pool/jammy/main/librealsense2-dkms_1.3.27-0ubuntu1_all.deb. The manual patching script is available at https://github.com/IntelRealSense/librealsense/blob/master/scripts/patch-realsense-ubuntu-lts-hwe.sh for Ubuntu LTS kernels, and at https://github.com/IntelRealSense/librealsense/blob/master/scripts/patch-linux-vanilla.sh for plain Linux kernels (but this script seems to be incorrect; it applies the PREEMPT_RT patch but not the patches for device drivers). The Ubuntu install guide using the Intel apt repository is at https://github.com/IntelRealSense/librealsense/blob/master/doc/distribution_linux.md, and the installation from source guide is available at https://github.com/IntelRealSense/librealsense/blob/master/doc/installation.md.

**Reply by ethanc8** [reply]
It looks like the patches are (ignoring those that only apply to very old kernels)
* realsense-camera-formats
* realsense-metadata (skip_md_patch) (use on kernel < 6.5)
* realsense-powerlinefrequency-control-fix (skip_plf_patch) (use on kernel < 6.0)
* uvcvideo_increase_UVC_URBS (skip_md_patch) (use on kernel < 6.5)
* uvc_debug (only use for debugging purposes, it adds additional logs)

**Comment by ethanc8** [comment]
Alternatively, we can use the RSUSB backend (which does not require kernel patching) instead: https://github.com/IntelRealSense/librealsense/issues/9931#issuecomment-964289692

**Comment by Joshua-Riek** [comment]
I can't use LTS kernels from Ubuntu, they do not support rk3588. You will need to port the driver proper.

**Reply by cmakelabs** [reply]
Thank you, not sure If I understood, I am using Radxa zero 3w with Ubuntu server(Joshua-Riek), could you please tell what do I need to do to run realsense on it?
